basics/getting_started/plugin_compatibility.md

```diff
@@ -2,80 +2,102 @@
 title: Plugin Compatibility with IntelliJ Platform Products
 ---
 
-<!--TODO link to sample_plugin file-->
+<style>
+  table {
+    width:100%;
+  }
+  th:first-child, td:first-child {
+    width: 30%;
+  }
+  th:last-child, td:last-child {
+    width: 25%;
+  }
+</style>
+
+## Introduction
+All products based on the _IntelliJ Platform_ are built on the same underlying API. 
+Some of these products share features built on top of the platform, such as Java support in IntelliJ IDEA and Android Studio. 
+Underlying those shared features are shared components. 
+When authoring a plugin for the IntelliJ Platform, it is important to understand and declare dependencies on these components. 
+Otherwise, it may not be possible to load or run the plugin in a product because the components on which it depends aren't available.
+
+## Declaring Plugin Dependencies
+For the purposes of dependencies, a _module_ can be thought of like a built-in plugin that ships as a non-removable part of a product. 
+A working definition of a dependency is that a plugin project cannot be run without the module present in an IntelliJ Platform-based product. 
+Declaring a dependency on a module also expresses a plugin's compatibility with a product in that the IntelliJ Platform determines whether a product contains the correct modules to support a plugin before loading it. 
+
+Note:
+* If a plugin does not include any module dependency tags in its `plugin.xml` file, it's assumed to be a legacy plugin and is loaded only in IntelliJ IDEA.
+  This form of dependency "non-declaration" is deprecated; do not use it for new plugin projects. 
```

Comment:
is that so? if yes, we must change our template `plugin.xml`

Reply:
I think so. I suggest to uncomment dependency on `com.intellij.modules.lang` in our template.

Reply:
Hmm, maybe it would be even better to use `com.intellij.modules.platform` as the default dependency (because `lang` module depends on `platform` module so `platform` module is more basic one).

Reply:
https://youtrack.jetbrains.com/issue/IDEA-221351